client/src/components/Sidebar/UserList/index.tsx

```diff
@@ -1,11 +1,25 @@
-import React, { useContext } from 'react'
-import { UserInfoContext } from "@/contexts";
+import React, { useContext, useState } from 'react'
+import { UserInfoContext, SocketContext } from "@/contexts";
 import { UserInfo } from "@/contexts/UserInfoContext";
 import AvatarDOM from "@/components/AvatarCanvas/AvatarDOM"
+import { SIOChannel } from "@/contexts/SocketIOContext";
 
 const UserList = () => {
   // TODO: Need to have selfUserInfo here as well
-  const { userInfos } = useContext(UserInfoContext);
+  const { userInfos, editUserName } = useContext(UserInfoContext);
```

Comment:
Looks like the bug where the userInfo gets reset is caused here:
You'll need to use `useRef` to persist the value between renders.
So it should look something like
```
const { userInfos } = useContext(UserInfoContext);
const userInfosRef = useRef(userInfos);
...
{(Object.values(userInfosRef.current) as...
```
You can also check the RoomPage for reference. We also do the same thing there.